apps/explorer/lib/explorer/chain/backward_transfer_decoding.ex

```diff
@@ -0,0 +1,60 @@
+defmodule BackwardTransfersDecoding do
+  @backward_transfer_contract_address <<
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 17, 17, 17, 17, 17, 17, 17, 17, 17, 17
+  >>
+
+  defp pub_key_hash_prefix do
+    subnetwork = System.get_env("SUBNETWORK") || ""
+    if String.contains?(subnetwork, "Testnet") do
+      <<32, 152>> # testnet, 2098
```

Comment:
My concern w this approach is that it leave room for human error, meaning, whoever is setting the SUBNETWORK env variable needs to know to include "Testnet" in the name.

Reply:
If there isn't a better way to use the existing env variables, adding `IS_MAINNET` works better.

Reply:
I also liked having the link to zencash here

Reply:
An alternative approach is 
`
if SUBNETWORK includes "EON" 
do mainnet 
else
do testnet
`

Reply:
I understood that we were ok with using the SUBNETWORK env variable. On my side, is just a matter to switching them, so I don't have a personal preference. For me, it works what works best for the team. The only thing is that adding new variables may involve infra adjusting the existing pipelines to include that variable as well.

@yurikoinaba Let me know if you think Shaun's suggestion is enough, or if you consider that involving infra on this is worth it.

Thank you!

Reply:
Let's add `IS_MAINNET`. Adding a new env variable shouldn't be a big lift on infra. Let's make sure to inform them.

To be safe, we can add a default value for the env variable if necessary

Reply:
Got it! The commit has been reverted, so we're back on using the `IS_MAINNET` variable.

@voyenavion do we need to modify any specific file for adding this new env to the pipeline? How should we notify infra about this change?

@yurikoinaba the `SUBNETWORK` env variable is used in multiple files (140 results in 25 files), so I guess is not only for displaying the chain name, so I agree it's better to not rely on it and use a specific variable like the originally proposed `IS_MAINNET`.

Reply:
Hi @jmontesdeocanuez, please add it with appropriate value to all of the env templates in this repo: https://github.com/HorizenLabs/compose-evm-blockscout/tree/main/env/eon-dev (you will prob need to get access, if you can request for the entire tools team git group) . Then create a ticket on the devops board for ensuring all the instances are updated. Finally, when the pr is merged post in the blockscout-infra-tools channel about the change.

Reply:
Hey @voyenavion, thank for your inputs!

I've created the ticket as requested. You can find it here: [Ticket #2212](https://horizenlabsinc.freshdesk.com/support/tickets/2212).

Regarding the task, I'm currently working on adding a default value to the `IS_MAINNET` variable. This change will ensure that if the variable is not defined, it will default to true, making the setup work as if it were Mainnet. Although this means that the decoded address might be incorrect on testnet if the variable isn't set, it will function correctly on the mainnet.

Reply:
lgtm

Reply:
<img width="868" alt="image" src="https://github.com/HorizenLabs/eon-explorer/assets/36055057/934bd63a-2999-4064-aa32-87faa76bbe82">
I don't have access for merging 🙏